components/layout/context.rs

```diff
@@ -119,32 +119,17 @@ impl<'a> LayoutContext<'a> {
             CanRequestImages::No
         };
 
-        // See if the image is already available
-        let result =
+        // Check for available image or start tracking.
+        let sender; // FIXME
```

Comment:
@jdm I left this broken for the moment. I wasn't sure where & how I should generate the `IpcSender` for the call to `track_image` here, since the `LayoutContext` isn't an element like the other callers of `track_image`. Based on the comments in the code above  I imagined what I would want to do would be to add a route to the `ipc_channel::router::ROUTER` with a callback that would trigger the reflow. However I also realized that it might make more sense to have the caller of `get_or_request_image_or_meta` pass a sender in. This would ultimately be `Fragement`s and `ImageFragmentInfo`s. any advice?

Reply:
Ok, I spent a while looking at this, and I think we don't actually want to call track_image here, because it doesn't really fit the model of the other consumers of that API. This code cares about three states:
1) is there an image/metadata already?
2) is there an image cache entry pending?
3) is there no image cache entry?

We only want to add a cache listener in states 2 and 3, and that listener is added in separate code in components/script/dom/window.rs (in Window::force_reflow). I think we should add a new image cache API called `get_cached_image_status` which returns ImageCacheResult and does not take a Sender argument. Hopefully we can find a good way to avoid duplicating the implementation of track_image.